### PR TITLE
Remove samples, recipes and outdated docs.

### PR DESCRIPTION
**Pull request by arturdryomov**
Superseded by recently rewritten documentation.

### CODE REVIEW

**Comment by arturdryomov** [comment]
@artem-zinnatullin, not sure why but GCC is required to assemble the binary.

```
Building debug version of Mainframer...
   Compiling mainframer v3.0.0-dev (/opt/project)
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)
```

**Comment by arturdryomov** [comment]
Also, had to explicitly install `rsync` since tests were not passing due to unknown executable. Seems like some package installed it implicitly.

File: ci/docker/Dockerfile [hunk 1]
```diff
@@ -5,17 +5,13 @@ MAINTAINER Mainframer Team
...
-# "lib32stdc++6 lib32z1 unzip": build sample Android project.
-# "ant python git": build sample Buck project.
+# "curl", "build-essential": build Mainframer.
+# "openssh-server", "rsync": testing.
 RUN apt-get update --quiet > /dev/null && \
-    apt-get --assume-yes --quiet install sudo openssh-server openjdk-8-jdk \
-    golang clang build-essential lib32stdc++6 lib32z1 unzip ant python git curl
+    apt-get --assume-yes --quiet install \
+    sudo \
+    curl build-essential \
```
Comment (artem-zinnatullin) [inline ci/docker/Dockerfile]: let's have all packages on separate lines

File: ci/docker/Dockerfile [hunk 2]
```diff
@@ -5,17 +5,13 @@ MAINTAINER Mainframer Team
...
+    openssh-server rsync
```
Comment (artem-zinnatullin) [inline ci/docker/Dockerfile]: Let's also run `apt-get clean` to cleanup intermediate files

**Review by artem-zinnatullin — APPROVED** [review]

**Comment by artem-zinnatullin** [comment]
Yeah Rust uses system linker to produce final binary (can be GCC, ld, or LLVMs lld afaik)

**Review by artem-zinnatullin — APPROVED** [review]

**Pull request merged**